src/api/firebase.js

```diff
@@ -116,17 +116,23 @@ export async function addUserToDatabase(user) {
  * @param {string} listName The name of the new list.
  */
 export async function createList(userId, userEmail, listName) {
-	const listDocRef = doc(db, userId, listName);
+	try {
+		const listDocRef = doc(db, userId, listName);
 
-	await setDoc(listDocRef, {
-		owner: userId,
-	});
+		await setDoc(listDocRef, {
+			owner: userId,
+		});
 
-	const userDocumentRef = doc(db, 'users', userEmail);
+		const userDocumentRef = doc(db, 'users', userEmail);
 
-	updateDoc(userDocumentRef, {
-		sharedLists: arrayUnion(listDocRef),
-	});
+		updateDoc(userDocumentRef, {
+			sharedLists: arrayUnion(listDocRef),
```

Comment:
Looked at the docs for [arrayUnion](https://cloud.google.com/firestore/docs/manage-data/add-data#update_elements_in_an_array) , I think both issues we've been working on this week refer to this page and I didn't even notice until poking around.

Reply:
Good find in the docs! We had figured out the piece about only adding elements not already present, but did it the hard way via trial/error.